### PR TITLE
Implements templated List and DropList classes.

### PR DESCRIPTION
**Pull request by tomsons26**

### CODE REVIEW

File: src/game/ui/tdroplist.h [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        DropButton = that.DropButton;
+        DropList = that.DropList;
+        DropButton.Make_Peer(*this);
+        DropList.Make_Peer(*this);
+    }
+
+    return *this;
+}
+
+//same as DropListClass
```
Comment (OmniBlade) [inline src/game/ui/tdroplist.h]: Don't put comments like this, put proper comment blocks and add an` @info Method implementation is the same as used in DropListClass` or something similar though that isn't all that useful to someone looking to use the classes API, only to someone investigating its internals.

File: src/game/ui/tdroplist.h [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+    virtual T Get_Item(int const string_index) const;
+
+    TDropListClass &operator=(TDropListClass &that);
+
+    void Expand();
+    void Collapse();
+
+protected:
+#ifndef CHRONOSHIFT_NO_BITFIELDS
+    BOOL IsExpanded : 1; // & 1
```
Comment (OmniBlade) [inline src/game/ui/tdroplist.h]: Use proper member format of `m_MemberName`.

File: src/game/ui/tlist.h [hunk 3]
```diff
@@ -0,0 +1,597 @@
...
+#else
+    bool HasScrollbar; // Does this List have a scroll bar for scrolling through the entries?
+#endif
+    ShapeButtonClass ScrollUpButton; // Scroll Up button instance for the scroll bar.
+    ShapeButtonClass ScrollDownButton; // Scroll Down button instance for the scroll bar.
+    SliderClass Scrollbar; // Scroll bar instance for the list.
+    int CurrentIndex;
+    int ViewIndex;
+};
+#include "gbuffer.h"
```
Comment (OmniBlade) [inline src/game/ui/tlist.h]: Don't put includes in the middle of the file.

**Review by OmniBlade — CHANGES_REQUESTED** [review]

**Review by OmniBlade — DISMISSED** [review]

**Review by ghost — DISMISSED** [review]

**Review by ghost — DISMISSED** [review]

File: src/game/ui/tdroplist.h [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+#else
+    bool m_IsExpanded;
+#endif
+    int m_DropHeight;
+    ShapeButtonClass m_DropButton;
+    TListClass<T> m_DropList;
+};
+
+template<typename T>
+inline TDropListClass<T> &TDropListClass<T>::operator=(TDropListClass<T> &that)
```
Comment (OmniBlade) [inline src/game/ui/tdroplist.h]: You don't need inline on templated functions, they are implicitly inline.

**Review by OmniBlade — CHANGES_REQUESTED** [review]

**Review by OmniBlade — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**